py-polars/polars/expr/expr.py

```diff
@@ -8038,7 +8038,7 @@ def shuffle(self, seed: int | None = None) -> Self:
 
     def sample(
         self,
-        n: int | None = None,
+        n: IntoExpr = None,
```

Comment:
I think `IntoExpr` is a bit to broad. Booleans, strings and dates don't make much sense here.